apps/wallet/src/ui/app/hooks/useDeepBook.ts

```diff
@@ -145,28 +153,151 @@ export function useBalanceConversion(
 
 	const rawValue = useMemo(() => {
-		if (!averagePrice || !balance) return null;
+		if (!averagePriceWithConversion || !balance) return null;
 
-		const rawUsdValue = new BigNumber(balance.toString()).multipliedBy(averagePrice).toNumber();
+		const rawUsdValue = new BigNumber(balance.toString())
+			.multipliedBy(averagePriceWithConversion)
+			.toNumber();
 
 		if (isNaN(rawUsdValue)) {
 			return null;
 		}
 
 		return rawUsdValue;
-	}, [averagePrice, balance]);
+	}, [averagePriceWithConversion, balance]);
 
 	return {
 		rawValue,
-		averagePrice,
+		averagePrice: averagePriceWithConversion,
 		...rest,
 	};
 }
 
 export function useSuiBalanceInUSDC(suiBalance: BigInt | BigNumber | null) {
-	return useBalanceConversion(suiBalance, Coins.SUI, Coins.USDC);
+	return useBalanceConversion(suiBalance, Coins.SUI, Coins.USDC, SUI_CONVERSION_RATE);
+}
+
+export function useCreateAccount() {
+	const activeAccount = useActiveAccount();
+	const signer = useSigner(activeAccount);
+
+	return useMutation({
+		mutationKey: [DEEPBOOK_KEY, 'create-account'],
+		mutationFn: async () => {
+			const deepBookClient = await getDeepBookClient();
+			const txb = deepBookClient.createAccount(activeAccount?.address);
+
+			return signer?.signAndExecuteTransactionBlock({ transactionBlock: txb });
+		},
+	});
+}
+
+export function useMarketAccountCap(activeAccountAddress?: string) {
+	const {
+		data,
+		isLoading: getOwnedObjectsLoading,
+		isError: getOwnedObjectsError,
+		isRefetching: isRefetchingOwnedObjects,
+		refetch,
+	} = useGetOwnedObjects(
+		activeAccountAddress,
+		{
+			MatchAll: [{ StructType: '0xdee9::custodian_v2::AccountCap' }],
+		},
+		10,
+	);
+
+	const objectId = data?.pages?.[0]?.data?.[0]?.data?.objectId;
+
+	const {
+		data: suiObjectResponseData,
+		isLoading: getObjectLoading,
+		isRefetching: isRefetchingObject,
+		isError: getObjectError,
+	} = useGetObject(objectId);
+
+	const fieldsData =
+		suiObjectResponseData?.data?.content?.dataType === 'moveObject'
+			? (suiObjectResponseData?.data?.content?.fields as Record<string, string | number | object>)
+			: null;
+
+	return {
+		data: fieldsData,
+		isLoading: getObjectLoading || getOwnedObjectsLoading,
+		isError: getOwnedObjectsError || getObjectError,
+		isRefetching: isRefetchingObject || isRefetchingOwnedObjects,
+		refetch,
+	};
+}
+
+export function useSuiUsdcTxnBlock(balanceInMist: string, accountCapId: string) {
+	const [txnBlock, setTxnBlock] = useState<TransactionBlock | undefined>();
+	const activeAccount = useActiveAccount();
+	const activeAccountAddress = activeAccount?.address;
+	const client = useSuiClient();
+
+	const { data: suiCoinsData } = useGetCoins(coinsMap.SUI, activeAccountAddress);
+	const { data: usdcCoinsData } = useGetCoins(coinsMap.USDC, activeAccountAddress);
+
+	const suiCoinObjectId = suiCoinsData?.pages?.[0]?.data?.[0]?.coinObjectId;
+	const usdcCoinObjectId = usdcCoinsData?.pages?.[0]?.data?.[0]?.coinObjectId;
+
+	useEffect(() => {
```

Comment:
Could this be a useQuery? Feels it would be simpler

Reply:
updated to `useQuery`